src/nvim/os/pty_conpty_win.c

```diff
@@ -168,6 +179,20 @@ void os_conpty_set_size(conpty_t *conpty_object, uint16_t width, uint16_t height
   assert(width <= SHRT_MAX);
   assert(height <= SHRT_MAX);
   COORD size = { (int16_t)width, (int16_t)height };
+  // is this pointer a valid pointer that was allocated from pCreatePseudoConsole
+  bool isValid = true;
+  if (pointerCount <= VALIDPOINTERSIZE) {
+    isValid = false;
+    for (int i = 0; i < pointerCount; i++) {
+      if (conpty_object->pty == validFilePointers[i]) {
```

Comment:
Elsewhere in this file I see checks for `INVALID_HANDLE_VALUE`, maybe that works instead?